frappe/core/doctype/docshare/docshare.py

```diff
@@ -40,6 +40,10 @@ def check_share_permission(self):
 			not frappe.has_permission(self.share_doctype, "share", self.get_doc())):
 
 			frappe.throw(_('You need to have "Share" permission'), frappe.PermissionError)
+			# if owner has the permission to write then only allow the self.user to write permission
```

Comment:
Hi, here self.owner is of document DocShare.
If user A creates a document and share with user B with (Read and Share permission) then User B will be able to share the document but he can not give the write permission.